cmd/producer/main_test.go

```diff
@@ -0,0 +1,135 @@
+/*
+Copyright 2020 The Knative Authors
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+package main
+
+import (
+	"context"
+	"errors"
+	"net/http"
+	"net/http/httptest"
+	"strings"
+	"testing"
+
+	"github.com/go-redis/redis/v8"
+)
+
+type fakeRedis struct {
+	client redis.Cmdable
+}
+
+func TestAsyncRequestHeader(t *testing.T) {
+	testserver := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		if r.Method != "GET" && r.Method != "POST" {
+			t.Errorf("Expected 'POST' OR 'GET' request, got '%s'", r.Method)
+		}
+	}))
+
+	tests := []struct {
+		name             string
+		async            bool
+		method           string
+		bodyType         string
```

Comment:
Can we pass the body here instead of bodyType? This way we can avoid the if statements to construct the body.

Reply:
yes, that would be better -- thanks!